kong/plugins/prometheus/exporter.lua

```diff
@@ -236,6 +271,28 @@ local function collect()
                                         {res.workers_lua_vms[i].pid})
   end
 
+  if metrics.config_current_hash then
+    local current_hash = declarative.get_current_hash()
+    -- current_hash is true if it's load from file
+    if current_hash and current_hash ~= true then
+      metrics.config_current_hash:set(config_hash_to_number(current_hash))
+    end
+  end
+
+  -- Hybrid mode status
+  if cp_metrics then
+    -- Cleanup old metrics
+    metrics.dataplane_last_seen:reset()
+    metrics.dataplane_config_hash:reset()
+
+    local data_planes = clustering.get_status()
+    for node_id, status in pairs(data_planes) do
+      local labels = { node_id, status.hostname, status.ip }
+      metrics.dataplane_last_seen:set(status.last_seen, labels)
+      metrics.dataplane_config_hash:set(config_hash_to_number(status.config_hash), labels)
+    end
+  end
```

Comment:
Please do not output the config hash of data planes on the control-plane. Output only the data-planes this control-plane is seeing and the config hash that it expects others to have.

The idea here is to give visibility of two kinds:
1. Control plane reports the data-plane it is seeing, this gives visibility into connections as they are working.
2. Control plane reports the desired hash and data-planes reports the current hash. The operator can then look for split-brains and such problems in here.

cc @wyndigo Does that align well?

Reply:
@hbagdi I'm okay to add additional config_hash on DP. There can be comparasion of mismatch on DP as well.
But I feel it's also important to gather it on CP. Consider use cases like user bring in their own DP, it will not be easy to ship metrics around. Actually for the current ways of gathering upstream/service metrics on DP feels odd to me. Ideally those states can be sync'ed across the cluster and at same time made available on CP.
Also like I proposed even with managed DP, it's likely deployed in different ways as CP and likely making labels inconsistent.

Reply:
Okay, I see the problem.

In that case, we should think about better names for the metrics.
`dataplane_config_hash` and `config_current_hash` communicate nothing about where the metrics are coming from or what they mean.

Reply:
I'm picking up this PR again as there're more metrics about hybrid mode I would like to
get in. @hbagdi Do you have suggestions on the naming here?